Sources/Hyperspace/Service/Transport/TransportService.swift

```diff
@@ -30,7 +30,7 @@ public class TransportService {
     }
     
     deinit {
-        cancelAllTasks()
+        tasks.forEach { $1.cancel() }
```

Comment:
It looks like there is an "issue" with the synchronous access in `deinit`. `self` is available before we synchronously enter the queue, but it is nil when we are in the queue, meaning the tasks aren't cancelled.  In the interest of not making sweeping changes in what is essentially a CI PR, I made this asynchronous call synchronous. @GrandLarseny, definitely worth looking at before we ship 4.0.0